packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1408,6 +1458,19 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 }
                 return create404Response(request);
             }
+
+            // The container runtime's entryPoint is now the object that should be able to handle requests;
+            // the requestHandler passed to the container runtime is now put in there
+            const resolvedEntryPoint = await this.entryPoint;
```

Comment:
Not sure I follow... I see this as the logic we need for now to ensure that `request()` on the runtime keeps working as it does today through the transition. The changes in this PR mean that the request handler people currently pass to `ContainerRuntime.load()` is not assigned to the runtime's `requestHandler` property anymore, but end up in the entryPoint. So for the whole thing to be non-breaking at the outset we need to keep that working how it does today without them having to make changes immediately.

Reply:
I'm suggesting to provide no out-of-the-box merging of the two paths but instead require the container author to stitch the two together before calling ContainerRuntime.load() if they need.  To do this you'd also add `initializeEntryPoint` to `.load()` so that callers can still pass requestHandler, entryPoint, or both as they see fit.  Any request handlers they pass continue to be used in the request() path same as today, no changes to that path at all.

If a container author needs back-compat for hosts calling `.request()`, they keep calling `.load()` with the same requestHandler as before but can also start passing the entry point that their hosts should migrate to using.  When back-compat is no longer needed (or when we remove `.request()`) they can stop passing the requestHandler and move to `.newLoad()`.

The two outcomes I think would be valuable to target:
1. Ensure ContainerRuntime does zero unpacking of the entryPoint (even through `IFluid*` probing).  By keeping it a total blackbox from the perspective of ContainerRuntime we avoid a class of maintainence/compat concerns.  We also don't have to document the magic behaviors in our public documentation.
    * As written, you probably need some line like "If you pass an entryPoint that also happens to be an `IFluidRouter`, then `IContainer.request()` will interrogate that `entryPoint` to respond."
2. Avoid patterns that encourage container authors to make their entry points be routers just for back-compat reasons.  Most entry points won't need to be routers, so prefer to let them write the entryPoint as they intend it to be long-term.  Instead do the back-compat support in the requestHandler, so that it can all be thrown away simultaneously when `.request()` support is dropped.
    * This probably looks like a request handler that maps `request({url: "foo"})` to `entryPoint.foo` or similar.

Reply:
Got you. Yeah, good feedback, thanks! We keep back-compat by just not touching the current flow for `request()`. If someone decides that their entryPoint will still be an `IFluidRouter`, that'll only matter when the hosts move to using the entryPoint and issuing requests directly to it, not through `IContainer.request()`. I'll make these changes.

Reply:
Addressed!